test/unit/storage_subformat_vagrant_libvirt_test.py

```diff
@@ -1,29 +1,16 @@
-import io
 from textwrap import dedent
-from mock import (
-    patch, Mock, MagicMock, call
-)
+from mock import patch, Mock, call
 
 from .test_helper import patch_open
+from .storage_subformat_vagrant_base_test import VagrantTestBase
 
 from kiwi.storage.subformat.vagrant_libvirt import DiskFormatVagrantLibVirt
 
 
-class TestDiskFormatVagrantLibVirt(object):
+class TestDiskFormatVagrantLibVirt(VagrantTestBase):
     def setup(self):
-        xml_data = Mock()
-        xml_data.get_name = Mock(
-            return_value='some-disk-image'
-        )
-        self.xml_state = Mock()
-        self.xml_state.xml_data = xml_data
-        self.xml_state.get_image_version = Mock(
-            return_value='1.2.3'
-        )
-        self.vagrantconfig = Mock()
-        self.vagrantconfig.get_virtualsize = Mock(
-            return_value=42
-        )
+        super(TestDiskFormatVagrantLibVirt, self).setup()
```

Comment:
ok, calling the base class constructor via super(). I know many people like this and I'm ok with it for the sake of handling the unit testing. So ok let's keep it.

```
class TestDiskFormatVagrantLibVirt(TestDiskFormatVagrantBase):
```

From an oo design perspective and the loosely coupled code idea, which I really prefer, super is like hell. That simple line creates a hard coupling to the base class and often opens the question if it really makes sense. The back call to the base on constructing usually leads to big headaches if there is any change to the base. It usually leads people to not touching the code because they fear the consequences. That's why I really appreciate what Sandy wrote in https://www.poodr.com/

Tests are also code, yes but at that level I accept a certain amount of code duplication. I have no problem writing a test and using the same init code again which I have used before in another test if I can avoid a coupling of test code. This is because I want to have full freedom to do changes in the implementation but if the tests start to try to become clever they will stand in my way for implementation refactoring.

It might sound boring or wrong to you but I have learned my lessons with super() :)